src/sql/src/plan/with_options.rs

```diff
@@ -180,6 +181,26 @@ impl ImpliedValue for Duration {
     }
 }
 
+impl TryFromValue<Value> for ByteSize {
+    fn try_from_value(v: Value) -> Result<Self, PlanError> {
+        match v {
+            Value::Number(value) | Value::String(value) => Ok(value
+                .parse::<ByteSize>()
+                .map_err(|e| sql_err!("invalid bytes value: {e}"))?),
+            _ => sql_bail!("cannot use value as bytes"),
+        }
+    }
```

Comment:
Not covered, can we add a test?

Reply:
Yeah, this is expected to be not covered. We don't have a use of this as a sql option yet. This will be used for the COPY to S3 sql. I can remove this, and make it part of the copy to s3 PR instead where it can be tested.

Reply:
No, it's okay!

Reply:
Since the other PR got merged first, I added a usage of this as a sql option. This part should now be covered by mzcompose test.